client/src/components/modals_and_popovers/ModalButton.tsx

```diff
@@ -1,13 +1,26 @@
 import React, { Fragment } from "react";
 
-import { withRouter } from "react-router-dom";
+import { withRouter, RouteComponentProps } from "react-router-dom";
 
 import { SafeJSURL } from "src/general_utils";
 
 import { StatelessModal } from "./StatelessModal";
 
-class ModalButton_ extends React.Component {
-  constructor(props) {
+type ModalButtonProps = RouteComponentProps & {
+  show_condition: { name: string; value: string };
+  match: { params: { options: string } };
+  button_text: string;
+  title: string;
+  children: string | React.ReactNode;
```

Comment:
`ReactNode` covers strings.

Here's a little pattern I use often to double check my assumptions about types:
![image](https://user-images.githubusercontent.com/11301919/126353113-afd027ea-df4b-44b6-b7d9-6b9ef117b926.png)

Reply:
That can be read as "can something of type `string` be stored in type `React.ReactNode`? `any` for true, `never` for false".